src/backend/main.ml

```diff
@@ -44,12 +38,28 @@ let () =
     |> Event.routes
     |> Competition.routes
   in
+  (* Define CORS middleware manually *)
+  let cors_middleware handler request =
+    match Dream.method_ request with
+    | `OPTIONS ->
+      Dream.respond ~headers:[
+        ("Access-Control-Allow-Origin", "*");
+        ("Access-Control-Allow-Methods", "GET, POST, PUT, DELETE, OPTIONS");
+        ("Access-Control-Allow-Headers", "Content-Type, Authorization");
+      ] ~status:`No_Content ""
+      | _ ->
```

Comment:
This is wrongly indented; you should install `ocp-indent` and use it to format your files, ^^